Testing

ashlynbaum opened the pull request:
Test API endpoints
List of endpoints for testing:
- [x] GET /events
- [x] GET /events/:id
- [x] DELETE /events/:id
- [x] POST /events
- [x] PATCH /events/:id
- [x] POST /signup
- [x] GET /      (login)
# TODO
- [x] Test all endoints
- [x] Setup Travis CI

Commit:
Configur mocha and add as dev dependancy

Because mocha is added to package.json, to test type ''$ nmp test''
to run the test.

Commit:
Test user sign up endpiont

Commit:
Add dev dependancy for testing

Commit:
Update README

Commit:
Add test for sign up and login endpoints

Commit:
Test basic auth user login endpoint

barodeur commented:
I forgot we still had the `POST /login` endpoint, but it's a useless endpoint since whe authenticate user through Basic Auth on `GET /` now.
Can you remove the endpoint on the master banch and then merge your master branch here ? ( so we don't lose the history of this nice pull request ).

Commit:
Fix style of tests

Commit:
Configure eslint

barodeur committed:
Fix some lint errors.

Commit:
Fix testing bug

Commit:
Remove POST /login endpoint replaced by GET /

Commit:
Fix linting errors.

Commit:
Add spec helper and refactor test into helper

Commit:
Change response from user sign up endpoint

Commit:
Begin test for creating events

barodeur committed:
Fix test that was using X-Auth-Token.

The new header for authorization is "Authorization" with the token scheme.

Commit:
Fix bug - server crashing on load.

Commit:
Refactor testing module and create event endpoint

Commit:
Update test create_event, fix bug in authorization middleware.

Commit:
Test event listing endpoint and update test for user sign up response.

Commit:
Modify error middleware to catch event thrown in GET /events/:id endpoint.

Error is thrown during call to mogoDB. Middleware now handles this error
and sends a status of 404.

Commit:
Test GET /events/:ID endpoint

Commit:
Test DELETE /events/:id

Commit:
Test PATCH /event/:id

Commit:
Update description of test DELETE

barodeur commented:
We can remove the `GET /test` endpoint and it's tests, I was just to try mocha :)

barodeur commented:
About the remaining lint errors, there are a lot of [`no-shadow`](http://eslint.org/docs/rules/no-shadow.html) usualy because of nesting callback that all have an `err` parameter. I think you can disabled the `no-shadow` check at the moment.

barodeur commented:
I'm removing the `GET /test`

barodeur committed:
Remove GET /test endpoint.

barodeur committed:
Capitalize "Get".

barodeur commented:
I just added a todo list item for this pull request wich consists on adding support for TravisCI. I'll explain you what is travis during our next call.

barodeur committed:
Separate databaseCleaner helper.

barodeur committed:
Refactor delete events tests.

Commit:
Remove no-shaddow on lint config

Commit:
Add request_app helper and modify DELETE test to incorporate module.

Commit:
Refactor test for listing events

Commit:
Refactor create event test

Commit:
Refactor show single event test

Commit:
Refactor update event test

Commit:
Fix typo

Commit:
Refactor show event test, remove unused variables

Commit:
Refactor user sign up test

Commit:
Refactor basic auth login test

Commit:
Refactor delete event test

Commit:
Delete init_db  helper

replaced functionality with database_Cleaner and request_app helper.

barodeur commented:
Great job @ashlynbaum last job is to setup travisCI now :)

Commit:
Add TravisCI continuous integration

Commit:
Add mongodb service for travisCI

Commit:
Update README add travisCI image
